components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/action/model/TokenRequest.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.oauth.action.model;
+
+import org.wso2.carbon.identity.action.execution.model.Request;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This class models the request at a pre issue access token trigger.
+ * TokenRequest is the entity that represents the request that is sent to Action
+ * over {@link org.wso2.carbon.identity.action.execution.model.ActionExecutionRequest}.
+ */
+public class TokenRequest extends Request {
+
+    private static final List<String> headersToAvoid = new ArrayList<>();
+    private static final List<String> paramsToAvoid = new ArrayList<>();
+
+    static {
+        headersToAvoid.add("authorization");
+        headersToAvoid.add("cookie");
+        headersToAvoid.add("set-cookie");
+        headersToAvoid.add("accept-encoding");
+        headersToAvoid.add("accept-language");
+        headersToAvoid.add("content-length");
+        headersToAvoid.add("content-type");
+        // parameters from authorization code grant
+        paramsToAvoid.add("code");
+        paramsToAvoid.add("client_id");
+        paramsToAvoid.add("client_secret");
+        paramsToAvoid.add("redirect_uri");
+        paramsToAvoid.add("grant_type");
+        paramsToAvoid.add("scope");
+        // parameters from password grant
+        paramsToAvoid.add("username");
+        paramsToAvoid.add("password");
+        // parameters from refresh token grant
+        paramsToAvoid.add("refresh_token");
```

Comment:
Missing sensitive params, `client_assertion`.

Reply:
When these values are taken into a file config level, we need to publish that set to the developers, in order to avoid any sensitive data leakage related security issues.

Reply:
Fixed